prime/src/components/graph/PnLGraph.tsx

```diff
@@ -252,23 +259,77 @@ export default function PnLGraph(props: PnLGraphProps) {
     setBorrowInterestInputValue,
     setSwapFeesInputValue,
   } = props;
+  const { activeChain } = useContext(ChainContext);
   const [data, setData] = useState<Array<PnLEntry>>([]);
   const [localInTermsOfToken0, setLocalInTermsOfToken0] = useState<boolean>(inTermsOfToken0);
+  const [priceAtLastUpdate, setPriceAtLastUpdate] = useState<number | null>(null);
 
-  let price = sqrtRatioToPrice(
+  useEffect(() => {
+    let mounted = true;
+    async function fetch() {
+      let etherscanResult: AxiosResponse<any, any> | null = null;
+      try {
+        etherscanResult = await makeEtherscanRequest(
+          0,
+          ALOE_II_FRONTEND_MANAGER_ADDRESS,
+          [TOPIC0_MODIFY_EVENT, `0x000000000000000000000000${marginAccount.address.slice(2)}`],
+          true,
+          activeChain
+        );
+      } catch (e) {
+        console.error(e);
+      }
+      if (etherscanResult == null || !Array.isArray(etherscanResult.data.result)) return [];
+      if (etherscanResult.data.result.length === 0) return [];
```

Comment:
Why return empty array here?

Reply:
Since if there are no events, we don't want to try to find the most recent